mesonbuild/compilers/vala.py

```diff
@@ -41,6 +46,22 @@ def get_output_args(self, outputname: str) -> T.List[str]:
     def get_compile_only_args(self) -> T.List[str]:
         return [] # Because compiles into C.
 
+    def get_compiler_args_for_mode(self, mode: CompileCheckMode) -> T.List[str]:
+        args: T.List[str] = []
+        if mode is CompileCheckMode.LINK and self.force_link:
+            return args
+        args += self.get_always_args()
+        if mode is CompileCheckMode.COMPILE:
+            args += self.get_compile_only_args()
+        elif mode is CompileCheckMode.PREPROCESS:
+            args += self.get_preprocess_only_args()
+        else:
+            assert mode is CompileCheckMode.LINK
```

Comment:
I don't think this is needed, the type checker should enforce this for us.

Reply:
Agreed.